homeassistant/components/bme280/sensor.py

```diff
@@ -1,135 +1,142 @@
 
+from .const import (
+    CONF_DELTA_TEMP,
+    CONF_FILTER_MODE,
+    CONF_I2C_ADDRESS,
+    CONF_I2C_BUS,
+    CONF_OPERATION_MODE,
+    CONF_OVERSAMPLING_HUM,
+    CONF_OVERSAMPLING_PRES,
+    CONF_OVERSAMPLING_TEMP,
+    CONF_SPI_BUS,
+    CONF_SPI_DEV,
+    CONF_T_STANDBY,
+    DEFAULT_SCAN_INTERVAL,
+    INTERFACE_I2C,
+    INTERFACE_SPI,
+    MIN_TIME_BETWEEN_UPDATES,
+    SENSOR_HUMID,
+    SENSOR_PRESS,
+    SENSOR_TEMP,
+    SENSOR_TYPES,
+)
+
 _LOGGER = logging.getLogger(__name__)
 
-CONF_I2C_ADDRESS = "i2c_address"
-CONF_I2C_BUS = "i2c_bus"
-CONF_OVERSAMPLING_TEMP = "oversampling_temperature"
-CONF_OVERSAMPLING_PRES = "oversampling_pressure"
-CONF_OVERSAMPLING_HUM = "oversampling_humidity"
-CONF_OPERATION_MODE = "operation_mode"
-CONF_T_STANDBY = "time_standby"
-CONF_FILTER_MODE = "filter_mode"
-CONF_DELTA_TEMP = "delta_temperature"
-
-DEFAULT_NAME = "BME280 Sensor"
-DEFAULT_I2C_ADDRESS = "0x76"
-DEFAULT_I2C_BUS = 1
-DEFAULT_OVERSAMPLING_TEMP = 1  # Temperature oversampling x 1
-DEFAULT_OVERSAMPLING_PRES = 1  # Pressure oversampling x 1
-DEFAULT_OVERSAMPLING_HUM = 1  # Humidity oversampling x 1
-DEFAULT_OPERATION_MODE = 3  # Normal mode (forced mode: 2)
-DEFAULT_T_STANDBY = 5  # Tstandby 5ms
-DEFAULT_FILTER_MODE = 0  # Filter off
-DEFAULT_DELTA_TEMP = 0.0
-
-MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=3)
-
-SENSOR_TEMP = "temperature"
-SENSOR_HUMID = "humidity"
-SENSOR_PRESS = "pressure"
-SENSOR_TYPES = {
-    SENSOR_TEMP: ["Temperature", None],
-    SENSOR_HUMID: ["Humidity", PERCENTAGE],
-    SENSOR_PRESS: ["Pressure", "mb"],
-}
-DEFAULT_MONITORED = [SENSOR_TEMP, SENSOR_HUMID, SENSOR_PRESS]
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS): cv.string,
-        vol.Optional(CONF_MONITORED_CONDITIONS, default=DEFAULT_MONITORED): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-        ),
-        vol.Optional(CONF_I2C_BUS, default=DEFAULT_I2C_BUS): vol.Coerce(int),
-        vol.Optional(
-            CONF_OVERSAMPLING_TEMP, default=DEFAULT_OVERSAMPLING_TEMP
-        ): vol.Coerce(int),
-        vol.Optional(
-            CONF_OVERSAMPLING_PRES, default=DEFAULT_OVERSAMPLING_PRES
-        ): vol.Coerce(int),
-        vol.Optional(
-            CONF_OVERSAMPLING_HUM, default=DEFAULT_OVERSAMPLING_HUM
-        ): vol.Coerce(int),
-        vol.Optional(CONF_OPERATION_MODE, default=DEFAULT_OPERATION_MODE): vol.Coerce(
-            int
-        ),
-        vol.Optional(CONF_T_STANDBY, default=DEFAULT_T_STANDBY): vol.Coerce(int),
-        vol.Optional(CONF_FILTER_MODE, default=DEFAULT_FILTER_MODE): vol.Coerce(int),
-        vol.Optional(CONF_DELTA_TEMP, default=DEFAULT_DELTA_TEMP): vol.Coerce(float),
-    }
-)
+SCAN_INTERVAL = timedelta(seconds=DEFAULT_SCAN_INTERVAL)
+
+
+def _set_scan_interval(interval: timedelta) -> None:
+    global SCAN_INTERVAL  # pylint: disable=global-statement
```

Comment:
Please don't use global. We don't support setting scan interval like this. If we need that option we need to use a different polling implementation, eg our update coordinator helper.

https://developers.home-assistant.io/docs/integration_fetching_data#coordinated-single-api-poll-for-data-for-all-entities

Reply:
Is it probably better to use Config Entries and use async_setup_entry instead of async_setup_platform? I need time to deal with all of these.

Reply:
Yes, I don't see a problem with introducing config entries. Maybe just do that refactor first instead?

Reply:
let's config entries leave until better times?